### PR TITLE
Appelflap canoe lock

### PR DESCRIPTION
**Pull request by md8n**
The very first thing that Canoe needs to do if it is running under Appelflap, is tell Appelflap to not reboot it until its actually ready.

### CODE REVIEW

File: src/riot/App.riot.html [hunk 1]
```diff
@@ -53,10 +53,21 @@
...
+        } from "js/Events";
+        import {
+            isBrowserSupported,
+            areCompletionsReady,
+            getManifestFromStore,
+        } from "ReduxImpl/Interface";
         import { logPageView } from "js/GoogleAnalytics";
 
+        import { getPlatform } from "js/PlatformDetection";
+        import { AppelflapConnect } from "ts/AppelflapConnect";
```
Comment (PeteCoward) [inline src/riot/App.riot.html:69]: Why not put this in the app.js? It seems totally unrelated to interface, so can be removed from this UI file.

File: src/app.js [hunk 2]
```diff
@@ -11,5 +14,34 @@ riot.install(function (component) {
...
+            riot.register("app", App);
+            riot.mount("app");
+        })
+        .catch((_) => {
+            /* Do nothing */
+        });
+} else {
+    riot.register("app", App);
+    riot.mount("app");
+}
```
Comment (PeteCoward) [inline src/app.js:47]: If it throws I'd rather not swallow it.
Using await avoids the repetition

```suggestion

    const lockResult = await lockCanoe();

    if (lockResult !== "ok") {
        // The lock didn't succeed, the app is probably still usable.
        // We don't yet understand what this means for the user and the app.
        // Should we show a small warning?
    }
}

riot.register("app", App);
riot.mount("app");
```

File: src/riot/App.riot.html (hunk 1 above)
Comment (md8n) [reply src/riot/App.riot.html]: I tried putting this in app.js - and it definitely wants riot running first ...

Comment (md8n) [reply src/riot/App.riot.html]: But I did extract a lot of the associated code out into its own little library, and then made corresponding changes

File: src/app.js (hunk 2 above)
Comment (md8n) [reply src/app.js]: This has all changed.

File: src/ts/CanoeHost.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                        // The lock didn't succeed, the app is probably still usable.
+                        // We don't yet understand what this means for the user and the app.
+                        // Should we show a small warning?
+                    }
+
+                    riotStart();
+                })
+                .catch(() => {
+                    /* Do nothing */
+                });
```
Comment (PeteCoward) [inline src/ts/CanoeHost.ts:26]: Why swallow exceptions like this?

Comment (PeteCoward) [reply src/ts/CanoeHost.ts:26]: the whole code is effectively
```
if (inAppelflap()) {
    await this.LockCanoe();
}
riotStart();
```
And you don't even need the riot start, so don't need the await

Comment (md8n) [reply src/ts/CanoeHost.ts:26]: Because we specifically do NOT want any exception here propagating, and therefore we must handle any exceptions even if that means just swallowing them.

This is a Node thing, but it's still general best practice for JS in the browser.

Comment (PeteCoward) [reply src/ts/CanoeHost.ts:26]: I specifically do want to propagate an exception thrown here ( unless you are _expecting_ the exception to happen regularly )

**Review by PeteCoward — APPROVED** [review]
